src/core/generators/angular.ts

```diff
@@ -53,14 +54,15 @@ ${
   headers?: HttpHeaders | {
       [header: string]: string | string[];
   };
-  observe?: any;
+  context?: HttpContext;
+  observe?: 'body' | 'events' | 'response';
   params?: HttpParams | {
-      [param: string]: string | string[];
+    [param: string]: string | number | boolean | ReadonlyArray<string | number | boolean>;
   };
   reportProgress?: boolean;
-  responseType?: any;
+  responseType?: 'arraybuffer' | 'blob' | 'json' | 'text';
```

Comment:
same as above

Reply:
ok reverted ;)

Reply:
Maybe the best approach would be to check the verb used by each path and generating the method name, use directly the type `Parameters<HttpClient['post']>`:
```javascript
  retrieveBrand<TData = BrandDto>(retrieveBrandBody: RetrieveBrandBody, options?: Parameters<HttpClient['post']>[2]): Observable<TData> {
    return this.http.post<TData>(`/api/Brand/RetrieveBrand`, retrieveBrandBody, options);
  }
```